contracts/staking/contracts/src/staking_pools/MixinStakingPoolRewards.sol

```diff
@@ -43,6 +46,26 @@ contract MixinStakingPoolRewards is
 
     using LibSafeMath for uint256;
 
+    /// @dev Syncs rewards for a delegator. This includes transferring rewards from
+    /// the Reward Vault to the Eth Vault, and adding/removing dependencies on cumulative rewards.
+    /// @param poolId Unique id of pool.
+    function syncDelegatorRewards(bytes32 poolId)
```

Comment:
Under what circumstances is this called? Who is expected to call it? It feels strange to have an external function callable by anyone that modifies state.

Reply:
This is used by a delegator when they want to sync their rewards without delegating/undelegating. It's effectively the same as delegating zero stake. 

Note also that this uses `msg.sender` and can't force a sync for an arbitrary delegator.